conference-sync/ConferenceSync.js

```diff
@@ -0,0 +1,150 @@
+/**
+ * Functions for syncing events from forms/spreadsheets to calendar entries.
+ *
+ * Original author: marcus@giantswarm.io
+ *
+ * Managed via: https://github.com/giantswarm/gapps-automation
+ */
+
+/** The prefix for properties specific to this script in the project. */
+const PROPERTY_PREFIX = 'ConferenceSync.';
+
+/** The Google Calendar to create conference events in. */
+const CALENDAR_ID_KEY = PROPERTY_PREFIX + 'calendarId';
+
+/** The sheet name inside the Google Sheets spreadsheet which serves as a container for this script project. */
+const SHEET_NAME_KEY = PROPERTY_PREFIX + 'sheetName';
+
+/** The URL of some kind of table or overview page or document. */
+const OVERVIEW_URL_KEY = PROPERTY_PREFIX + 'overviewUrl';
+
+/** A Slack app bot token.
+ *
+ * The connected Slack app must:
+ *   - have at least the scopes channels:read, groups:read, chat:write and chat:write.public
+ *   - be added to a private or public channel to which notifications will be posted
+ */
+const SLACK_BOT_TOKEN_KEY = PROPERTY_PREFIX + 'slackBotToken';
+
+/** Call for papers deadline warning threshold in days (default: 7). */
+const DEADLINE_WARNING_DAYS = PROPERTY_PREFIX + 'deadlineWarnDays';
+
+/** Register this function with a "On form submit" trigger in Sheet context.
+ *
+ * USAGE:
+ *
+ *  1. Deploy this script into a spreadsheet with the right columns (conference form)
+ *  2. Ensure the calendar is shared with and subscribed by the account running the script
+ *  3. Set the property ConferenceSync.calendarId to the calendar ID.
+ *  4. Setup "On form submit" trigger to call onFormSubmit()
+ */
+function onFormSubmit() {
+    const calendarId = getScriptProperties_().getProperty(CALENDAR_ID_KEY);
+    if (!calendarId) {
+        throw new Error('Script property ConferenceSync.calendarId not configured, please set to target calendar ID');
+    }
+
+    const spreadsheet = SpreadsheetApp.getActiveSpreadsheet();
+    if (!spreadsheet) {
+        throw new Error('No active spreadsheet, please deploy this script into a spreadsheet context');
+    }
+
+    const sheetName = getScriptProperties_().getProperty(SHEET_NAME_KEY);
+    syncToCalendar_(spreadsheet.getId(), sheetName, calendarId);
+}
+
+/** Synchronize rows from a sheet with well-known columns from a conference form to a calendar. */
+function syncToCalendar_(spreadsheetId, sheetName, calendarId) {
```

Comment:
Why do these functions all have a `_` suffix?

Reply:
This is App Script's way to define which functions are public (part of the interface, callable via Execution API/Web) and which are "private" (not part of the interface or directly callable from outside the script context.